### PR TITLE
Issue#180

### PR DESCRIPTION
**Pull request by tusharravindran**
Resolve Issue#180 Only Moderator can edit their Community , when other users try to access flash message notice "Unauthorised access to Edit"

### CODE REVIEW

**Comment by tusharravindran** [comment]
Conflict resolved @iMagesh

File: app/controllers/communities_controller.rb [hunk 1]
```diff
@@ -1,70 +1,98 @@
...
-      @communities = Community.where(category: params[:category]).order(created_at: :desc).page(params[:page]).per 7
-    else
-      @communities = Community.order(created_at: :desc).page(params[:page]).per 7
-    end
-    @random_category_communities = Community.where(category: @categories.sample).order("post_count_this_week desc").limit(5)
+    @communities = if params.has_key?(:category)
+                     Community.where(category: params[:category]).order(created_at: :desc).page(params[:page]).per 7
+                   else
+                     Community.order(created_at: :desc).page(params[:page]).per 7
+                   end
```
Comment (iMagesh) [inline app/controllers/communities_controller.rb:15]: there are some alignment issues here? why is this happening? @tusharravindran

File: app/controllers/communities_controller.rb [hunk 2]
```diff
@@ -1,70 +1,98 @@
...
+    @community = Community.find(params[:id])
+    @posts = @community.posts.limit(20).sort_by { |p| p.score }.reverse
     @subscriber_count = @community.subscribers.count
-    @subscribed = is_subscribed?
+    @subscribed = if account_signed_in?
+                    Subscription.where(community_id: @community.id,
+                                       account_id: current_account.id).any?
+                  else
+                    false
+                  end
```
Comment (iMagesh) [inline app/controllers/communities_controller.rb:28]: see this? why is this happening?
Please make sure you don't mess up the code like this

File: app/controllers/communities_controller.rb [hunk 3]
```diff
@@ -1,70 +1,98 @@
...
   end
 
   def mod
-    unless @community.account_id == current_account.id
-      redirect_back(fallback_location: root_path) and return
-    end
-    @banned_user = set_banned_user.order(created_at: :desc).page(params[:page]).per 5
+    redirect_back(fallback_location: root_path) and return unless @community.owner_id == current_account.id
+
+    @banned_user = BannedUser.where(community_id: @community.id).order(created_at: :desc).page(params[:page]).per 5
```
Comment (iMagesh) [inline app/controllers/communities_controller.rb:79]: Check this as well

File: app/controllers/public_controller.rb [hunk 4]
```diff
@@ -20,7 +20,7 @@ def profile
     @posts = @profile.posts
     @my_comments = Comment.where(account_id: @profile.id)
     @my_posts = Post.where(account_id: @profile.id, is_drafted: false).page(params[:page]).per 5
-    @hot_myposts = Post.where(account_id: @profile.id).order(view_count: :desc).page(params[:page]).per 5
+    @hot_myposts = Post.where(account_id: @profile.id).order(upvotes: :desc).page(params[:page]).per 5
     @top_myposts = Post.where(account_id: @profile.id).order(view_count: :desc).page(params[:page]).per 5
     @new_myposts = Post.where(account_id: @profile.id).order(created_at: :desc).page(params[:page]).per 5
```
Comment (iMagesh) [inline app/controllers/public_controller.rb:25]: Can you refactor this, please? use separate partials for each tab and also do it via ajax @tusharravindran

File: config/locales/flash.en.yml [hunk 5]
```diff
@@ -1,6 +1,7 @@
 en:
   form:
     required: "Fill all required fields"
+    forbidden: "Unauthorised access to Edit"
```
Comment (iMagesh) [inline config/locales/flash.en.yml:4]: Can you change the message?

forbidden: You don't have permission to edit

File: app/controllers/communities_controller.rb (hunk 1 above)
Comment (tusharravindran) [reply app/controllers/communities_controller.rb:15]: This is due to refactoring code using the RuboCop gem .I will resolve it now

File: app/controllers/communities_controller.rb (hunk 2 above)
Comment (tusharravindran) [reply app/controllers/communities_controller.rb:28]: This is due to refactoring code using the RuboCop gem .I will resolve it now

File: config/locales/flash.en.yml (hunk 5 above)
Comment (tusharravindran) [reply config/locales/flash.en.yml:4]: Ok @iMagesh

File: app/controllers/public_controller.rb (hunk 4 above)
Comment (tusharravindran) [reply app/controllers/public_controller.rb:25]: In Progress .Will Resolve it soon

File: app/controllers/communities_controller.rb [hunk 6]
```diff
@@ -8,21 +8,20 @@ class CommunitiesController < ApplicationController
...
     @community = Community.find(params[:id])
     @posts = @community.posts.limit(20).sort_by { |p| p.score }.reverse
     @subscriber_count = @community.subscribers.count
     @subscribed = if account_signed_in?
-                    Subscription.where(community_id: @community.id,
-                                       account_id: current_account.id).any?
+                    Subscription.where(community_id: @community.id,account_id: current_account.id).any?
                   else
                     false
                   end
```
Comment (iMagesh) [inline app/controllers/communities_controller.rb:27]: the alignment is still off here.. check

Comment (tusharravindran) [reply app/controllers/communities_controller.rb:27]: image
Is this not a proper alignment or should I refactor it differently @iMagesh

File: app/controllers/public_controller.rb (hunk 4 above)
Comment (tusharravindran) [reply app/controllers/public_controller.rb:25]: Can we do it as a seperate PR @iMagesh due to same type of code in [Index]
image

Comment (tusharravindran) [reply app/controllers/public_controller.rb:25]: I will do it as a separate Branch and send PR

File: app/controllers/communities_controller.rb (hunk 6 above)
Comment (iMagesh) [reply app/controllers/communities_controller.rb:27]: this is not a proper alignment @tusharravindran

Comment (tusharravindran) [reply app/controllers/communities_controller.rb]: Resolved @iMagesh

**Pull request merged**